tools/generate_type_hints.py

```diff
@@ -0,0 +1,231 @@
+#!/usr/bin/env python
+"""A script for generating stub files with platform-specific types."""
+
+import os
+import sys
+import argparse
+import ctypes as ct
+from collections import defaultdict
+from typing import Union, Dict, IO
+
+__all__ = ["generate_alias"]
+
+_AnyPath = Union[str, bytes, os.PathLike]
+
+ANNOTATED_CHARCODES = {
+    "int8",
+    "int16",
+    "int32",
+    "int64",
+    "uint8",
+    "uint16",
+    "uint32",
+    "uint64",
+    "float16",
+    "float32",
+    "float64",
+    "complex64",
+    "complex128",
+}
+
+ALLOWED_SIZES = {8, 16, 32, 64, 80, 96, 128, 256}
+
+#: A mapping with `number` aliases as keys and the name of their
+#: character-code-containing `Literal` as values.
```

Comment:
If you're going to go down the route of using ctypes, this needs a clear comment about why you're doing it.

Reply:
The various comments and docstrings have been updated as of https://github.com/numpy/numpy/pull/17514/commits/9cfcea012ec17eb4c4e27480901c1ef82a8a2e20; it should now be clearer what exactly is going on with the `ctypes` approach.